File: src/harkerrobolib/util/Gains.java
package harkerrobolib.util;

/**
 * Designed to store PID constants in an easily accessible way.
 * @author Finn Frankis
 * @author Shahzeb Lakhani
 * @author Anirudh Kotamraju
 * 
 * @version April 25, 2020
 */
public class Gains {
    private double kF;
    private double kP;
    private double kI;
    private double kD;
    private int iZone;
	
    public Gains kF (double kF) { this.kF = kF; return this;}
    public Gains kP (double kP) {this.kP = kP; return this;} 
    public Gains kI (double kI) {this.kI = kI; return this;}
    public Gains kD (double kD) {this.kD = kD; return this;}
    public Gains iZone (int iZone) {this.iZone = iZone; return this;}
	
    public double getkF() {return kF;}

	public double getkP() {return kP;} 

	public double getkI() {return kI;}

	public double getkD() {return kD;}
	
	public int getIZone() {return iZone;}
}

File: src/harkerrobolib/util/Gains.kt
package harkerrobolib.util

/**
 * Designed to store PID constants in an easily accessible way.
 * @author Finn Frankis
 * @version 10/21/18
 */
data class Gains (val kF: Double = 0.0, val kP: Double = 0.0, val kI: Double = 0.0, val kD: Double = 0.0, val iZone: Double = 0.0)
